conformance/lang/expressions/literals/floating_point_literal.balt

```diff
@@ -0,0 +1,5665 @@
+}
+
+Test-Case: error
+Description: Test the basic type when the floating point literal includes a DecimalTypeSuffix.
+Labels: FloatingPointTypeSuffix, float, int
+
+function errorFunction() {
+    int _ = 25d; // @error expected an 'int' but found a 'decimal'
+    int _ = 25D; // @error expected an 'int' but found a 'decimal'
+    float _ = .99E-1742D; // @error expected a 'float' but found a 'decimal'
+    float _ = 245.32E-1742D; // @error expected a 'float' but found a 'decimal'
+    float _ = 25.0E+1742d; // @error expected a 'float' but found a 'decimal'
+    float _ = .99E1742D; // @error expected a 'float' but found a 'decimal'
+    float _ = 245.32E1742D; // @error expected a 'float' but found a 'decimal'
+    float _ = 25.0E1742d; // @error expected a 'float' but found a 'decimal'
+    float _ = .99d; // @error expected a 'float' but found a 'decimal'
+    float _ = 245.32D; // @error expected a 'float' but found a 'decimal'
+    float _ = 25.0d; // @error expected a 'float' but found a 'decimal'
+    float _ = 25E-1742D; // @error expected a 'float' but found a 'decimal'
+    float _ = 25E1742d; // @error expected a 'float' but found a 'decimal'
+}
+
+Test-Case: error
+Description: Test the basic type when the floating point literal is a HexFloatingPointLiteral
+Labels: decimal, HexFloatingPointLiteral, int
+
+function errorFunction() {
+    decimal _ = 0x3eA3p34; // @error expected 'decimal', found 'float'
+    decimal _ = 0x3eA3p+34; // @error expected 'decimal', found 'float'
+    decimal _ = 0X3eA3P-34; // @error expected 'decimal', found 'float'
+    decimal _ = 0x3eA3.4Eb2; // @error expected 'decimal', found 'float'
+    decimal _ = 0x.4Eb2; // @error expected 'decimal', found 'float'
+    decimal _ = 0x4Eb2.; // @error expected 'decimal', found 'float'
+    decimal _ = 0x3eA3.4Eb2p45; // @error expected 'decimal', found 'float'
+    decimal _ = 0x.4Eb2p45; // @error expected 'decimal', found 'float'
+    decimal _ = 0x3eA3.4Eb2p+45; // @error expected 'decimal', found 'float'
+    decimal _ = 0x.4Eb2p-45; // @error expected 'decimal', found 'float'
+    int _ = 0x3eA3p34; // @error expected 'int', found 'float'
+}
+
+Test-Case: output
+Description: Test if +0 and -0 can be distinguished.
+Labels: decimal, float, is-expr
+
+function init() {
+    io:println(+0 is -0); // @output true
+    io:println(+0 == -0); // @output true
+    io:println(+0 != -0); // @output false
+    io:println(+0f is -0f); // @output true
+    io:println(+0f == -0f); // @output true
+    io:println(+0f != -0f); // @output false
+    io:println(+0d is -0d); // @output true
+    io:println(+0d == -0d); // @output true
+    io:println(+0d != -0d); // @output false
+}
+
+Test-Case: output
+Description: Test DecimalFloatingPointNumber with valid type cast to integer type.
+Labels: type-cast-expr, FloatingPointTypeSuffix, int
+
+function init() {
+
+    io:println(<int> 25e8); // @output 2500000000
```

Comment:
Since we've added these, it's OK to keep. But these would be a better fit for type cast tests.